src/addon.cpp

```diff
@@ -80,15 +80,76 @@ extern "C" {
 
 NAN_MODULE_INIT(init)
 {
+#ifdef _WIN32
+  Nan::Set(target, JS_STR("WIN"), Nan::True());
+  Nan::Set(target, JS_STR("UNIX"), Nan::False());
+  Nan::Set(target, JS_STR("POSIX"), Nan::False());
+  Nan::Set(target, JS_STR("LINUX"), Nan::False());
+  Nan::Set(target, JS_STR("APPLE"), Nan::False());
+
+#elif defined(__APPLE__) || defined(MACOSX)
+  Nan::Set(target, JS_STR("WIN"), Nan::False());
+  Nan::Set(target, JS_STR("UNIX"), Nan::False());
+  Nan::Set(target, JS_STR("POSIX"), Nan::False());
+  Nan::Set(target, JS_STR("LINUX"), Nan::False());
+  Nan::Set(target, JS_STR("APPLE"), Nan::True());
+#elif __linux__
+  // linux
+  Nan::Set(target, JS_STR("WIN"), Nan::False());
+  Nan::Set(target, JS_STR("UNIX"), Nan::False());
+  Nan::Set(target, JS_STR("POSIX"), Nan::False());
+  Nan::Set(target, JS_STR("LINUX"), Nan::True());
+  Nan::Set(target, JS_STR("APPLE"), Nan::False());
+#elif __unix__ // all unices not caught above
+  // Unix
+  Nan::Set(target, JS_STR("WIN"), Nan::False());
+  Nan::Set(target, JS_STR("UNIX"), Nan::True());
+  Nan::Set(target, JS_STR("POSIX"), Nan::False());
+  Nan::Set(target, JS_STR("LINUX"), Nan::False());
+  Nan::Set(target, JS_STR("APPLE"), Nan::False());
+#elif defined(_POSIX_VERSION)
+  Nan::Set(target, JS_STR("WIN"), Nan::False());
+  Nan::Set(target, JS_STR("UNIX"), Nan::False());
+  Nan::Set(target, JS_STR("POSIX"), Nan::True());
+  Nan::Set(target, JS_STR("LINUX"), Nan::False());
+  Nan::Set(target, JS_STR("APPLE"), Nan::False());
+#else
+#  error "Unknown compiler"
+#endif
+
+
+Nan::Set(target, JS_STR("version"), JS_STR("1.0"));
+Nan::Set(target, JS_STR("v10"), Nan::True());
+
+#ifdef CL_VERSION_1_1
+  Nan::Set(target, JS_STR("version"), JS_STR("1.1"));
+  Nan::Set(target, JS_STR("v11"), Nan::True());
+#else
+  Nan::Set(target, JS_STR("v11"), Nan::False());
+#endif
 #ifdef CL_VERSION_1_2
-  Nan::Set(target, JS_STR("CL_VERSION_1_2" ), Nan::True());
+  Nan::Set(target, JS_STR("version"), JS_STR("1.2"));
+  Nan::Set(target, JS_STR("v12"), Nan::True());
 #else
-  Nan::Set(target, JS_STR("CL_VERSION_1_2" ), Nan::False());
+  Nan::Set(target, JS_STR("v12"), Nan::False());
```

Comment:
@lu4 any particular reason these were renamed? In general I'd be happy to merge this PR, but I think we should avoid changing things in the public API unless there's a good reason.